staging/src/k8s.io/kms/encryption/service_test.go

```diff
@@ -361,29 +437,331 @@ func TestDecryptError(t *testing.T) {
 }
 
 func TestStatus(t *testing.T) {
+	ctx := testContext(t)
 	remoteKMS := &testRemoteService{keyID: "test-key-id"}
-	localKEKService := NewLocalKEKService(remoteKMS)
+	localKEKService := newLocalKEKService(ctx, remoteKMS, 10, 5, 500*time.Millisecond, 100*time.Millisecond)
 
-	ctx := testContext(t)
+	waitUntilReady(t, localKEKService)
+
+	validateResponse := func(got *service.StatusResponse, wantVersion, wantHealthz, wantKeyID string, t *testing.T) {
```

Comment:
As a general rule, please avoid closures like this - they can easily get out of hand and are much harder to compose - simply create regular functions and use them for test assertions.